src/test/java/io/cdap/plugin/servicenow/source/ServiceNowMultiSourceConfigTest.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.servicenow.source;
+
+import io.cdap.cdap.etl.api.validation.ValidationFailure;
+import io.cdap.cdap.etl.mock.validation.MockFailureCollector;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.util.List;
+
+/**
+ * Tests for {@link ServiceNowMultiSourceConfig}.
+ */
+public class ServiceNowMultiSourceConfigTest {
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+  private ServiceNowMultiSourceConfig serviceNowMultiSourceConfig;
+
+  @Before
+  public void initializeTests() {
+    serviceNowMultiSourceConfig =
+      ServiceNowSourceConfigHelper.newConfigBuilder()
+        .setReferenceName("Reference Name")
+        .setRestApiEndpoint("https://example.com")
+        .setUser("user")
+        .setPassword("password")
+        .setClientId("client_id")
+        .setClientSecret("client_secret")
+        .setTableNames("sys_user")
+        .setValueType("Actual")
+        .setStartDate("2021-12-30")
+        .setEndDate("2021-12-31")
+        .setTableNameField("tablename")
+        .buildMultiSource();
+  }
+
+  @Test
+  public void testConstructor() {
+    Assert.assertEquals("sys_user", serviceNowMultiSourceConfig.getTableNames());
+    Assert.assertEquals("Actual", serviceNowMultiSourceConfig.getValueType().getValueType());
+    Assert.assertEquals("2021-12-30", serviceNowMultiSourceConfig.getStartDate());
+    Assert.assertEquals("2021-12-31", serviceNowMultiSourceConfig.getEndDate());
+    Assert.assertEquals("tablename", serviceNowMultiSourceConfig.getTableNameField());
+  }
+
+  @Test
+  public void testValidate() {
+    MockFailureCollector mockFailureCollector = new MockFailureCollector("Stage Name");
+    serviceNowMultiSourceConfig.validate(mockFailureCollector);
+    Assert.assertEquals(3, mockFailureCollector.getValidationFailures().size());
```

Comment:
why are there 3 failures? Generally test 1 thing at a time. It should be clear what you're testing from reading the test code. Also, prefer the initialize serviceNowMultiSourceConfig in the same method for better readability

Reply:
The test case is being re-written to have only 1 assertion. `serviceNowMultiSourceConfig` will be initialized in the test method itself. The refactored test case will be included in the next PR for service-now JUnit.